validation/src/test/scala/hmda/validation/rules/lar/macro/Q071Spec.scala

```diff
@@ -0,0 +1,169 @@
+      val numSold = (200 * (0.6 - yearDifference) - 1).toInt
+      val relevantNotSoldLars = listOfN(200 - numSold, Q071Spec.relevantNotSold)
+      val relevantSoldLars = listOfN(numSold, Q071Spec.relevantSold)
+      val irrelevantLars = listOfN(any, Q071Spec.irrelevant)
+      val testLars = toSource(relevantSoldLars ++ relevantNotSoldLars ++ irrelevantLars)
+      Q071.inContext(ctx(instId))(testLars).map(r => r mustBe a[Failure])
+    }
+
+    //// Check #2: Current Year ////
+    "(current year check) fails when too few relevant loans sold to Ginnie Mae" in {
+      val instId = "first"
+      val numSold: Int = (threshold * proportionSold).toInt
+      val relevantNotSoldLars = listOfN(threshold - numSold, Q071Spec.relevantNotSold)
+      val relevantSoldLars = listOfN(numSold, Q071Spec.relevantSold)
+      val irrelevantLars = listOfN(any, Q071Spec.irrelevant)
+      val testLars = toSource(relevantSoldLars ++ relevantNotSoldLars ++ irrelevantLars)
+      Q071.inContext(ctx(instId))(testLars).map(r => r mustBe a[Failure])
+    }
+    "(current year check) passes when enough relevant loans sold to Ginnie Mae" in {
+      val instId = "second"
+      val numSold: Int = (threshold * proportionSold).toInt + 1
+      val relevantNotSoldLars = listOfN(threshold - numSold, Q071Spec.relevantNotSold)
+      val relevantSoldLars = listOfN(numSold, Q071Spec.relevantSold)
+      val irrelevantLars = listOfN(any, Q071Spec.irrelevant)
+      val testLars = toSource(relevantSoldLars ++ relevantNotSoldLars ++ irrelevantLars)
+      Q071.inContext(ctx(instId))(testLars).map(r => r mustBe a[Success])
+    }
+    s"(current year check) passes when number of relevant loans is below $threshold" in {
+      val instId = "third"
+      val relevantSoldLars = listOfN(threshold - 1, Q071Spec.relevantSold)
+      val irrelevantLars = listOfN(any, Q071Spec.irrelevant)
+      val testLars = toSource(relevantSoldLars ++ irrelevantLars)
+      Q071.inContext(ctx(instId))(testLars).map(r => r mustBe a[Success])
+    }
+    s"(current year check) doesn't blow up when there are 0 relevant loans" in {
+      val instId = "fourth"
+      val irrelevantLarSource = toSource(listOfN(any, Q071Spec.irrelevant))
+      Q071.inContext(ctx(instId))(irrelevantLarSource).map(r => r mustBe a[Success])
+    }
+
+    //// Must handle context correctly ////
+    "be named Q071 when institution and year are present" in {
+      Q071.inContext(ctx("Any")).name mustBe "Q071"
+    }
+
+    "be named 'empty' when institution or year is not present" in {
+      val ctx1 = ValidationContext(None, Some(currentYear))
+      Q071.inContext(ctx1).name mustBe "empty"
+
+      val ctx2 = ValidationContext(Some(Institution.empty), None)
+      Q071.inContext(ctx2).name mustBe "empty"
+    }
+
+  }
+
+  private def ctx(institutionId: String, currentYear: Int = 2017): ValidationContext = {
```

Comment:
Looks like a lot of these helper methods are going to be shared between several other tests... Would it be possible to move them into a separate file?